tests/test_validation.py

```diff
@@ -0,0 +1,135 @@
+    def test_warn_unphysical_values(self, validation_mode, caplog):
+        nan_tfs_path = INPUTS_DIR / "has_nans.tfs"
+        _ = read_tfs(nan_tfs_path, index="NAME", validate=validation_mode)
+        for record in caplog.records:
+            assert record.levelname == "WARNING"
+        assert "contains non-physical values at Index:" in caplog.text
+
+    def test_warning_on_non_unique_columns(self, caplog):
+        df = TfsDataFrame(columns=["A", "B", "A"])
+        validate(df)
+
+        for record in caplog.records:
+            assert record.levelname == "WARNING"
+        assert "Non-unique column names found" in caplog.text
+
+    def test_warning_on_non_unique_index(self, caplog):
+        df = TfsDataFrame(index=["A", "B", "A"])
+        validate(df)
+
+        for record in caplog.records:
+            assert record.levelname == "WARNING"
+        assert "Non-unique indices found" in caplog.text
+
+    def test_warning_on_non_unique_both(self, tmp_path, caplog):
+        df = TfsDataFrame(index=["A", "B", "A"], columns=["A", "B", "A"])
+        validate(df)
+
+        for record in caplog.records:
+            assert record.levelname == "WARNING"
+        assert "Non-unique column names found" in caplog.text
+        assert "Non-unique indices found" in caplog.text
+
+
+class TestCommonFailures:
+    """Tests for common failures in validation, both in MAD-X and MAD-NG compatibility mode."""
+
+    @pytest.mark.parametrize("validation_mode", ["madx", "mad-x", "madng", "MAD-NG"])
+    def test_validate_raises_on_wrong_unique_behavior(self, validation_mode):
+        df = TfsDataFrame(index=["A", "B", "A"], columns=["A", "B", "A"])
+        with pytest.raises(ValueError):
+            validate(df, "", non_unique_behavior="invalid", compatibility=validation_mode)
+
+    @pytest.mark.parametrize("validation_mode", ["madx", "mad-x", "madng", "MAD-NG"])
+    def test_validation_raises_space_in_colname(self, _space_in_colnames_tfs_path: pathlib.Path, validation_mode):
+        # Read file has a space in a column name which should raise
+        with pytest.raises(SpaceinColumnNameError, match="TFS-Columns can not contain spaces."):
+            _ = read_tfs(_space_in_colnames_tfs_path, index="NAME", validate=validation_mode)
+
+    @pytest.mark.parametrize("validation_mode", ["madx", "mad-x", "madng", "MAD-NG"])
+    def test_validation_raises_wrong_boolean_header(self, _invalid_bool_in_header_tfs_file, validation_mode):
+        # The file header has a boolean value that is not accepted
+        with pytest.raises(InvalidBooleanHeaderError, match="Invalid boolean header value parsed"):
+            _ = read_tfs(_invalid_bool_in_header_tfs_file, validate=validation_mode)
+
+    @pytest.mark.parametrize("validation_mode", ["not ok", "ma-Dx", "nope", "madngg"])
+    def test_validation_raises_on_invalid_compatibility_mode(self, _tfs_dataframe, validation_mode):
```

Comment:
with my suggestion of allowing falsy values as validation modes in reader/writer, maybe also add two test.raises for `_ = read_tfs` and `_ = write_tfs` into this test

Reply:
Following our discussion and the change, only `None` is accepted to skip validation in reader and writer, and checks for validity are done by the validate function. Still want a test? Or did I misunderstand?

Reply:
Yes, you can add this to the test that you already have that should raise the errors, only you call not only the validate function directly, but also the read and write function, which then should go into the validate function to raise the same error.

Reply:
Oooh okay yes, will do.